src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -493,17 +493,33 @@ internal static string TupleMemberName(int position)
             return "Item" + position;
         }
 
+        // This array needs to be sorted for binary searching
+        private static readonly string[] reservedMemberNames = new[]
+        {
```

Comment:
Why do we need to hardcode the names? Can we just ask underlying type for member dictionary and check if if key exist in the dictionary?